src/inspector_profiler.cc

```diff
@@ -503,6 +507,217 @@ static void StopCoverage(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+void ProfileWatchdog::Start() {
+  if (state() == ProfileWatchdogState::kInitialized) {
```

Comment:
I think you need to check if `state() !== kRunning` otherwise it will create a second thread and eventually cause dead locks.

Reply:
That still isn't enough because there's a race window between the check and the state update. Atomic operations don't cut it, it needs to be a critical section (i.e., a mutex.)

(Basically a rehash of what https://github.com/nodejs/node/pull/45714/files#r1045100682 points out.)

Reply:
Got it. Indeed, mutexes seem appropriated then.

Reply:
The writer is main thread, the reader is worker thread, only the main thread will update the `state`, do we need a mutex ?

Reply:
If that's not easily reproducible (as any deadlock), I'd say to keep it without mutex. If someone suffers from deadlock, we update it. How does it sound to you @bnoordhuis?

Reply:
Skirting correctness in multi-threaded code doesn't pay off.